mentat/mentat_prompt_session.py

```diff
@@ -200,7 +202,10 @@ def prompt(self, *args, **kwargs):
         while (user_input := super().prompt(*args, **kwargs)).startswith("/"):
             arguments = shlex.split(user_input[1:])
             command = Command.create_command(arguments[0])
-            command.apply(*arguments[1:])
+            if isinstance(command, (AddCommand, RemoveCommand)):
+                command.apply(*arguments[1:], code_context=self.code_context)
```

Comment:
i think this is a perfect spot to use **kwargs; we can always pass in code_context, no matter what the command type, and add and remove can just access it with kwargs["code_context"]; let me know what you think, but I think this is cleaner and makes it a lot easier in the future to add other possible singletons that are needed.

Reply:
I actually had done this initially, but opted to have `code_context` specified directly for commands that need it. Purely a personal preference, but my thinking is it's clearer what's going on when we explicitly define method arguments.

Reply:
Hmm... I like **kwargs a lot more since it means we don't need this isinstance here (I think the actual implementation of the commands should be only in commands file, and it shouldn't have to bleed out into other files like this). @biobootloader, what do you think?

Reply:
Ah that makes sense. In that case I think either choice is good as long as we're consistent :)

Reply:
I agree that `**kwargs` obscures what's going on, so I usually don't like it, but also it's a bit awkward to have to check. I think I have an idea to simplify let me try it out and commit here

Reply:
Ok I changed it so we pass `code_context` to `create_command`. This way we don't need `**kwargs` and don't need to check command type outside of the commands file. Commit here: https://github.com/biobootloader/mentat/pull/88/commits/61c10536dd996c7d0398e3216961d811c51545d6